### PR TITLE
feat: Ec2 2023 section9 sls

### PR DESCRIPTION
**Pull request by aperezcaldentey**
First partial version of Section 9 or FprEN 1992-1-1:2022

### CODE REVIEW

**Comment by mortenengen** [comment]
I am sorry. I accidentally closed the PR when I deleted the branch in #7...

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,6 @@
     "python.testing.unittestEnabled": false,
     "python.testing.pytestEnabled": true,
     "python.linting.pylintEnabled": true,
-    "python.linting.flake8Enabled": true
```
Comment (DanielGMorenaFhecor) [inline .vscode/settings.json:9]: Please, leave settings.json files with the default configuration

File: structuralcodes/codes/ec2_2023/_section9_sls.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    kb: float,
+    phi: float,
+    rho_eff: float,
+    x: float,
+    sigma_s: float,
+    kt: float,
+    fct_eff: float,
+    alphae: float,
+    Es: float,
+):
```
Comment (DanielGMorenaFhecor) [inline structuralcodes/codes/ec2_2023/_section9_sls.py:211]: Please, specify the return type -> ```-> Tuple[float, float, float, float]```

**Review by DanielGMorenaFhecor — CHANGES_REQUESTED** [review]
Good work! Please, take a look at the comments and tell me if you have any suggestion!

File: tests/test_ec2_2023_section9_sls.py
Comment (DanielGMorenaFhecor) [inline tests/test_ec2_2023_section9_sls.py:1]: A relative tolerance (`rel_tol`) of 0.05 means that your equality tests will return `true` if your values have a difference lower or equal than 5%, which is pretty high. 

Take into consideration that the default value is `rel_tol=10e-9`, so something about 10e-5 should be fine!

File: structuralcodes/codes/ec2_2023/_section5_materials.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+"""Functions from Section 5 of FprEN 1992-1-1:2022"""
+
+import math
+
+from structuralcodes.codes import mc2010
+
+# import structuralcodes.codes.mc2010 as mc2010
```
[the remark this answers is not in the document]
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section5_materials.py]: Done

File: structuralcodes/codes/ec2_2023/_section5_materials.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def fcm(fck: float, delta_f: float = 8.0) -> float:
```
[the remark this answers is not in the document]
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section5_materials.py:8]: Docstrings regarding Args, Kwargs and return values added where missing

File: structuralcodes/codes/ec2_2023/_section9_sls.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
+"""Functions from Section 9 of FprEN 1992-1-1:2023"""
+
+
+def Ec_eff(fcm: float, phi: float, kE: float = 9500) -> float:
+    """Returns de Effective modulus of elasticity from fcm and phi
```
[the remark this answers is not in the document]
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:11]: Corrected

File: structuralcodes/codes/ec2_2023/_section9_sls.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    alphae: float,
+    Es: float,
+) -> float:
+    """Returns the mean strain difference between steel and concrete along
+    2 transfer lengths
+
+    FprEN 1992-1-1:2023 Eq. (9.11)
+
+    Args:
+        sigmas (float): the stress in steel at the section of the crack
```
[the remark this answers is not in the document]
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:138]: Corrected

File: structuralcodes/codes/ec2_2023/_section9_sls.py (hunk 2 above)
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:217]: Done

File: structuralcodes/codes/ec2_2023/_section5_materials.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    Returns:
+        float: The mean tensile strength in MPa.
+    """
+    if abs(fck) <= 50:
+        return 0.3 * math.pow(abs(fck), 2 / 3)
+    return 1.1 * math.pow(abs(fck), 1 / 3)
+
+
+def Ecm(fcm1: float, kE: float = 9500) -> float:
```
Comment (DanielGMorenaFhecor) [inline structuralcodes/codes/ec2_2023/_section5_materials.py:45]: I think that docstring regarding kE is missing here

File: structuralcodes/codes/ec2_2023/_section9_sls.py [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+    """Returns k1/r factor to account for increase in crack width due to
+    curvature of the section in bending.
+
+    EN 1992-1-1:2023 Eq. (9.9)
+
+    Args:
+        h (float): height of the section in consistent units (e.g. meters)
+        x (float): distance from most compressed fibre to neutra axis in
+        consistent units (e.g. meters)
+        ay: cover to centre of tensioned reinforcement closest to most
```
Comment (DanielGMorenaFhecor) [inline structuralcodes/codes/ec2_2023/_section9_sls.py:116]: ay -> ay (float)

File: structuralcodes/codes/ec2_2023/_section9_sls.py [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+    EN 1992-1-1:2023 Eq. (9.8)
+
+    Args:
+        kw: factor that converts the mean crack spacing to a characteristic
+        value
+        k_1_r_: factor accounting for the effect of curvature on cracl width -
+                can be determined using the function k_1_r
+        srm_cal_: mean crack spacing - can be determines using the function
+                  srm_cal
+        epssm_epscm_: mean diference of strain between steel anc concrete - can
```
Comment (DanielGMorenaFhecor) [inline structuralcodes/codes/ec2_2023/_section9_sls.py:100]: missing 'Returns' docstring

File: structuralcodes/codes/ec2_2023/_section9_sls.py
Comment (DanielGMorenaFhecor) [inline structuralcodes/codes/ec2_2023/_section9_sls.py:1]: After reviewing this file, I noticed that some Returns statements in the Docstrings are missing. Please, tell me if you want me to add them!

**Review by DanielGMorenaFhecor — CHANGES_REQUESTED** [review]
Nice work! I think that still some Docstrings are missing (my fault, as I did not realized the last time). Please, tell my if you want me to add them and I'll do it 👍

File: structuralcodes/codes/ec2_2023/_section9_sls.py
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:1]: Thank you Daniel, I will include the missing definitions

File: structuralcodes/codes/ec2_2023/_section9_sls.py (hunk 8 above)
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:116]: done

File: structuralcodes/codes/ec2_2023/_section5_materials.py (hunk 7 above)
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section5_materials.py:45]: done

File: structuralcodes/codes/ec2_2023/_section9_sls.py (hunk 9 above)
Comment (aperezcaldentey) [reply structuralcodes/codes/ec2_2023/_section9_sls.py:100]: Added "Returns:
        the crack width in in units consistent with srm_cal"

**Review by DanielGMorenaFhecor — APPROVED** [review]

**Comment by DanielGMorenaFhecor** [comment]
I have just updated some changes so there are no more conflicts with the most recent `dev` branch changes! Nice work!

**Comment by DanielGMorenaFhecor** [comment]
I have just reviewed the code so it is aligned with the typing practices we specified in our last message!

**Comment by mortenengen** [comment]
Great @DanielGMorenaFhecor! I will have a look and merge if no further changes are required.

**Pull request merged**